File: src/services/projectService.js
import { api } from './api'

export const projectService = {
  async getProjects() {
    const { data } = await api.get('/projects')
    return data
  },

  async getProject(id) {
    const { data } = await api.get(`/projects/${id}`)
    return data
  },

  async createProject(payload) {
    const { data } = await api.post('/projects', payload)
    return data
  },

  async updateProject(id, payload) {
    const { data } = await api.put(`/projects/${id}`, payload)
    return data
  },

  async deleteProject(id) {
    await api.delete(`/projects/${id}`)
  },

  async inviteMember(projectId, payload) {
    const { data } = await api.post(`/projects/${projectId}/invite`, payload)
    return data
  },
}

File: src/main/java/com/taskflow/service/ProjectService.java
package com.taskflow.service;

import com.taskflow.dto.InviteMemberRequest;
import com.taskflow.dto.ProjectRequest;
import com.taskflow.dto.ProjectResponse;
import java.util.List;
import org.springframework.security.core.Authentication;

public interface ProjectService {
    ProjectResponse createProject(ProjectRequest request, Authentication authentication);

    List<ProjectResponse> getProjects(Authentication authentication);

    ProjectResponse getProject(String id, Authentication authentication);

    ProjectResponse updateProject(String id, ProjectRequest request, Authentication authentication);

    void deleteProject(String id, Authentication authentication);

    ProjectResponse inviteMember(String projectId, InviteMemberRequest request, Authentication authentication);
}
